### PR TITLE
Add custom mutator that receives seeds from llm.

### PR DESCRIPTION
**Pull request by harrison4ride**
1. Add a new custom mutator that receives seeds from llm.
2. Enable send and receive messages between fuzzer and LLM.
3. Add a new bitmap to evaluate the seeds from LLM.
4. Send the evaluated reward back to LLM.

### CODE REVIEW

File: custom_mutators/aflpp/aflpp.c [hunk 1]
```diff
@@ -20,58 +56,97 @@ my_mutator_t *afl_custom_init(afl_state_t *afl, unsigned int seed) {
...
+      if (my_msg.data_type == TYPE_SEED){
+        printf("RECEIVED");
+        size_t hexLength = strlen(my_msg.data_buff);
+        size_t byteLength = hexLength / 2;
+        if (MAX_FILE < byteLength) {
+          printf("Buffer size %ld is too small for the hex string %ld.\n",MAX_FILE,byteLength);
+          break;
+        }
+
+        for (size_t i = 0, j = 0; i < hexLength; i += 2, j++) {
```
Comment (DataCorrupted) [inline custom_mutators/aflpp/aflpp.c:106]: Why don't you just send a buffer of bytes?

File: custom_mutators/aflpp_baseline/aflpp.c [hunk 2]
```diff
@@ -0,0 +1,100 @@
+// This simple example just creates random buffer <= 100 filled with 'A'
```
Comment (DataCorrupted) [inline custom_mutators/aflpp_baseline/aflpp.c]: Please comment, what is this file for?

File: custom_mutators/aflpp_baseline/aflpp.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+typedef struct my_mutator {
+
+  afl_state_t *afl;
+
+  // Reused buffers:
+  u8 *fuzz_buf;
+
+} my_mutator_t;
+
+int hexCharToVal(char c) {
```
Comment (DataCorrupted) [inline custom_mutators/aflpp_baseline/aflpp.c]: Is  this function useful?

**Review by DataCorrupted — CHANGES_REQUESTED** [review]

**Comment by DataCorrupted** [comment]
Please don't PR to stable as well. You can PR it to something like `LLM-stable`

File: custom_mutators/aflpp/aflpp.c (hunk 1 above)
Comment (harrison4ride) [reply custom_mutators/aflpp/aflpp.c:106]: It is send a buffer of bytes, and this part of codes is to write hex string to the seed buffer.

File: custom_mutators/aflpp_baseline/aflpp.c (hunk 3 above)
Comment (harrison4ride) [reply custom_mutators/aflpp_baseline/aflpp.c]: Yes, it writes hex string to the seed buffer.

File: custom_mutators/aflpp/aflpp.c [hunk 4]
```diff
@@ -1,17 +1,53 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/msg.h>
+#include <time.h>
+
+#ifndef _FIXED_CHAR
+  #define _FIXED_CHAR 0x41
```
Comment (DataCorrupted) [inline custom_mutators/aflpp/aflpp.c]: Please comment or write documentations about what is `FIXED_CHAR`, what is it for, why is it `0x41`.

File: custom_mutators/aflpp/aflpp.c [hunk 5]
```diff
@@ -20,58 +56,96 @@ my_mutator_t *afl_custom_init(afl_state_t *afl, unsigned int seed) {
...
-
 size_t afl_custom_fuzz(my_mutator_t *data, uint8_t *buf, size_t buf_size,
-                       u8 **out_buf, uint8_t *add_buf, size_t add_buf_size,
+                       u8 **out_buf, uint8_t *add_buf,
+                       size_t add_buf_size,  // add_buf can be NULL
                        size_t max_size) {
 
-  if (max_size > data->buf_size) {
+  int size = (rand() % 100) + 1;
+  if (size > max_size) size = max_size;
```
Comment (DataCorrupted) [inline custom_mutators/aflpp/aflpp.c]: 1. rand() is bad [[1]](https://stackoverflow.com/questions/52869166/why-is-the-use-of-rand-considered-bad) [[2]](https://www.youtube.com/watch?v=LDPMpc-ENqY)
2. If you need a number [0-99], why three is a `max_size`? Why don't you get a number [0-max_size] at the first place?

File: custom_mutators/aflpp_baseline/aflpp.c (hunk 3 above)
Comment (DataCorrupted) [reply custom_mutators/aflpp_baseline/aflpp.c]: It is not used anywhere in this file.

File: custom_mutators/aflpp/aflpp.c (hunk 1 above)
Comment (DataCorrupted) [reply custom_mutators/aflpp/aflpp.c:106]: Why sending "ff" (two bytes) when you can send just one byte with value of 255? You are spending double the memory and extra time to decode "ff" to 255.

**Review by DataCorrupted — CHANGES_REQUESTED** [review]

**Pull request merged**